syntax = "proto3";
package iroha.protocol;

message AddAssetQuantity {
    string account_id = 1;
    string asset_id = 2;
    string amount = 3;
}

message AddPeer {
    string address = 1;
    bytes peer_key = 2;
}

message AddSignatory {
    string account_id = 1;
    bytes public_key = 2;
}

message AssignMasterKey {
    string account_id = 1;
    bytes public_key = 2;
}

message CreateAsset {
    string asset_name = 1;
    string domain_id = 2;
    uint32 precision = 3;
}

message CreateAccount {
    string user_name = 1;
    string domain_id = 2;
    bytes main_pubkey = 3;
}

message CreateDomain {
    string domain_name = 1;
}

message RemoveSignatory {
    string account_id = 1;
    bytes public_key = 2;
}

message SetAccountPermissions {
    string account_id = 1;
    uint32 permissions = 2;
}

message SetAccountQuorum {
    string account_id = 1;
    uint32 quorum = 2;
}

message TransferAsset {
    string src_account_id = 1;
    string dest_account_id = 2;
    string asset_id = 3;
    string ammount = 4;
}

message Command {
    oneof command {
        AddAssetQuantity add_asset_quantity = 1;
        AddPeer add_peer = 2;
        AddSignatory add_signatory = 3;
        AssignMasterKey account_assign_mk = 4;
        CreateAsset create_asset = 5;
        CreateAccount create_account = 6;
        CreateDomain create_domain = 7;
        RemoveSignatory remove_sign = 8;
        SetAccountPermissions set_permission = 9;
        SetAccountQuorum    set_quorum = 10;
        TransferAsset   transfer_asset = 11;
    }
}
